src/Policies.cpp

```diff
@@ -276,6 +276,28 @@ std::vector<Policy *> PolicyManager::load(const YAML::Node &policy_yaml)
     return result;
 }
 
+void PolicyManager::remove_policy(const std::string &name)
+{
+    std::unique_lock lock(_map_mutex);
+    if (_map.count(name) == 0) {
+        throw ModuleException(name, fmt::format("module name '{}' does not exist", name));
+    }
+
+    auto policy = _map[name].get();
+    std::string input_name = policy->input_stream()->name();
+    std::vector<std::string> module_names;
+    for (auto &mod : policy->modules()) {
+        module_names.push_back(mod->name());
+    }
+    policy->stop();
+    _map.erase(name);
+    lock.unlock();
```

Comment:
@weyrick I've applied the same previous logic here to avoid locking HandlerManger and InputManager mutexes inside PolicyManager. here we could erase from map only at the end and keep everything in PolicyManger lock context. What do you think is safer?

Reply:
Yes let's unlock at the end - otherwise we may have a race if the policy is recreated quickly and the handler or input names collide